libs/brig-types/src/Brig/Types/Common.hs

```diff
@@ -111,3 +121,74 @@ instance FromJSON ExcludedPrefix where
 
 instance ToJSON ExcludedPrefix where
   toJSON (ExcludedPrefix p c) = object ["phone_prefix" .= p, "comment" .= c]
+
+-------------------------------------------------------------------------------
+-- Unique Keys
+
+-- | An 'EmailKey' is an 'Email' in a form that serves as a unique lookup key.
+data EmailKey = EmailKey
+  { emailKeyUniq :: !Text,
+    emailKeyOrig :: !Email
+  }
+
+instance Show EmailKey where
+  showsPrec _ = shows . emailKeyUniq
+
+instance Eq EmailKey where
+  (EmailKey k _) == (EmailKey k' _) = k == k'
+
+-- | Turn an 'Email' into an 'EmailKey'.
+--
+-- The following transformations are performed:
+--
+--   * Both local and domain parts are forced to lowercase to make
```

Comment:
While it isn't common, the local part is "domain dependent" and can be case-sensitive.  We probably shouldn't case fold it by default, though we can if we know the domain does handle the local part case-insensitively.

Case-folding by default will make for some very hard to trace issues if the domain does use case-sensitive local parts.

Reply:
Just like the rest in this module, this is not new code added by me in the PR; rather, it's just being reorganised and moved from other modules. That said, I don't think this PR should be about addressing your concern and I believe you should open a ticket and bring it to the team's attention.